lib/futurism/resolver/controller/renderer.rb

```diff
@@ -0,0 +1,55 @@
+module Futurism
+  module Resolver
+    class Controller
+      class Renderer
+        def self.for(controller:, connection:, url:, params:)
+          new(controller: controller, connection: connection, url: url, params: params).renderer
+        end
+
+        def initialize(controller:, connection:, url:, params:)
+          @controller = controller
+          @connection = connection
+          @url = url || ""
+          @params = params || {}
+
+          setup_env!
+        end
+
+        def renderer
+          @renderer ||= controller.renderer
+        end
+
+        private
+
+        attr_reader :controller, :connection, :url, :params
+        attr_writer :renderer
+
+        def setup_env!
+          if url.present?
+            uri = URI.parse(url)
+            path = ActionDispatch::Journey::Router::Utils.normalize_path(uri.path)
+            query_hash = Rack::Utils.parse_nested_query(uri.query)
```

Comment:
I'm a bit wary if this could be used to inject bad code into the server... but I guess it's the same risk as with any other request?

Reply:
hmm. I'm not sure. Can you give an example?

Reply:
You know, sql injection etc. 

But I guess the risk is the same as with any standard get request. Just making sure we don’t overlook something.